lib/src/controls/navigation/tree_view.dart

```diff
@@ -309,12 +365,39 @@ class TreeView extends StatefulWidget {
   /// Called when an item is invoked
   final Future<void> Function(TreeViewItem item)? onItemInvoked;
 
+  /// Called when the selection changes. The items that are currently
+  /// selected will be passed to the callback. This could be empty
+  /// if nothing is now selected. If [TreeView.selectionMode] is
+  /// [TreeViewSelectionMode.single] then it will contain exactly
+  /// zero or one items.
+  final Future<void> Function(Iterable<TreeViewItem> selectedItems)?
+      onSelectionChanged;
+
   /// A widget to be shown when a node is loading. Only used if
   /// [TreeViewItem.loadingWidget] is null.
   ///
   /// [kTreeViewLoadingIndicator] is used by default
   final Widget loadingWidget;
 
+  /// Same meaning as [ListView.shrinkWrap]
+  final bool shrinkWrap;
+
+  /// Same meaning as [ListView.cacheExtent]
+  final double? cacheExtent;
+
+  /// Same meaning as [ListView.itemExtent]
+  final double? itemExtent;
+
+  /// Same meaning as [ListView.addRepaintBoundaries]
```

Comment:
```suggestion
  /// Whether to wrap each child in a [RepaintBoundary].
  ///
  /// Typically, children in a scrolling container are wrapped in repaint
  /// boundaries so that they do not need to be repainted as the list scrolls.
  /// If the children are easy to repaint (e.g., solid color blocks or a short
  /// snippet of text), it might be more efficient to not add a repaint boundary
  /// and simply repaint the children during scrolling.
  ///
  /// Defaults to true.
```